File: perl6/sample/class_defines/around_alias.pl
#!/usr/bin/env perl6

use v6;

use MONKEY_TYPING;

augment class Str {
    .^add_method('real_chars', &chars);

    method chars {
        .real_chars > 5 ?? 'long' !! 'short'
    }
}

say 'War and Peace'.chars;      # => 'long'
say 'War and Peace'.real_chars; # => 13

File: ruby/sample/class_defines/around_alias.rb
#!/usr/bin/env ruby19

class String
  alias :real_length :length

  def length
    real_length > 5 ? 'long' : 'short'
  end
end

p 'War and Peace'.length      #=> 'long'
p 'War and Peace'.real_length #=> 13
